gammapy/stats/li_ma.py

```diff
@@ -0,0 +1,347 @@
+
+    Note: The required step in log likelihood for "nsig sigma errors" is offset = 0.5*n_sig^2
+            http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/node30.html
+
+    Parameters
+    ----------
+    n_on : array_like
+        Observed number of counts in the on region
+    n_off : array_like
+        Observed number of counts in the off region
+    alpha : array_like
+        On / off region exposure ratio for background events
+    n_sig : float
+        Number of sigma
+
+    Returns
+    -------
+    significance : `~numpy.ndarray`
+        Error estimate
+
+    References
+    ----------
+    .. [1] Li and Ma, "Analysis methods for results in gamma-ray astronomy",
+       `Link <https://ui.adsabs.harvard.edu/abs/1983ApJ...272..317L>`_
+    """
+    scalar = False
+    if np.isscalar(n_on):
+        n_on = np.array([n_on])
+        scalar = True
+    if np.isscalar(n_off):
+        n_off = np.array([n_off])
+        scalar = True
+    if np.isscalar(alpha):
+        alpha = np.array([alpha])
+        scalar = True
+    n_on = np.asarray(n_on, dtype=np.float64)
+    n_off = np.asanyarray(n_off, dtype=np.float64)
+    alpha = np.asanyarray(alpha, dtype=np.float64)
+    n_sig = np.full(n_on.shape, n_sig)
+
+    # Optimal value of signal
+    excess0 = n_on - n_off * alpha
+    l0 = np.zeros(n_on.shape)
+    for ii in range(len(n_on)):
+        l0[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], excess0[ii])
+
+    # Search for signal such as -2 * log H > 1 * sig
+    # i.e. find a signal > dExcess_Up so that we can
+    # use an interval bisection algorithm to find dExcess_Up
+    offset = np.full(n_on.shape, n_sig*n_sig*0.5)
+    signal = excess0.copy() + 1
+    ll = np.zeros(n_on.shape)
+    for ii in range(len(ll)):
+        ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+        count = 0
+        while (l0[ii]-ll[ii]) < offset[ii] and np.isfinite(ll[ii]) and count < 100:
+            signal[ii] = (signal[ii]-excess0[ii]) * 2. + excess0[ii]
+            ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+            count += 1
+
+    # Dichotomic search (i.e. interval bisection algorithm)
+    low = excess0.copy()
+    up = signal.copy()
+    signal = (up+low)/2.
+    for ii in range(len(ll)):
+        ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+        count = 0.
+        while np.abs(l0[ii]-ll[ii]-offset[ii]) > 1e-4 and np.isfinite(ll[ii]) and count < 100:
+            if (l0[ii]-ll[ii]) > offset[ii]:
+                up[ii] = signal[ii]
+            else:
+                low[ii] = signal[ii]
+            signal[ii] = (up[ii]+low[ii])/2.
+            ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+            count += 1
+
+    if scalar:
+        return np.abs(signal[0] - excess0[0])
+    return np.abs(signal - excess0)
+
+
+def lm_dexcess(n_on, n_off, alpha, n_sig=1.):
```

Comment:
My recommendation would be to just remove this function. I think it's totally fine if users write the 3 lines to compute the mean themselves. I think there is not need to introduce an extra function with a lot of documentation. for such a simple operation...